lib/puppet/functions/archive/go_md5.rb

```diff
@@ -0,0 +1,32 @@
+# Public: go file md5 checksum
+#
+# args[0] - username
+# args[1] - password
+# args[2] - file_name
+# args[3] - go md5 checksum url
+#
+# http://www.thoughtworks.com/products/docs/go/12.4/help/Artifacts_API.html
+#
+# Returns specific file's md5 from go server md5 checksum file
+
+Puppet::Functions.create_function(:'archive::go_md5') do
+  require 'puppet_x/bodeco/util'
+
+  dispatch :main do
+    required_param 'String', :username
+    required_param 'String', :password
+    required_param 'String', :file
+    required_param 'String', :url
```

Comment:
Should this use `Stdlib::Httpurl` or are protocols other than http(s) also supported?

Reply:
It checked against String in define archive::nexus anyway